src/task_manager.cc

```diff
@@ -119,9 +119,9 @@ namespace detail {
 			if(std::any_of(modes.cbegin(), modes.cend(), detail::access::mode_traits::is_producer) || reduction.has_value()) {
 				auto write_requirements = get_requirements(tsk.get(), bid, {detail::access::producer_modes.cbegin(), detail::access::producer_modes.cend()});
 				if(reduction.has_value()) { write_requirements = GridRegion<3>::merge(write_requirements, GridRegion<3>{{1, 1, 1}}); }
-				assert(!write_requirements.empty() && "Task specified empty buffer range requirement. This indicates potential anti-pattern.");
-				const auto last_writers = buffers_last_writers.at(bid).get_region_values(write_requirements);
+				if(write_requirements.empty()) continue;
```

Comment:
I am torn on this as well. Specifying an empty requirement explicitly is probably a user error for which I would like to print a warning. If an empty requirement originates from generic code though, it might be a necessary corner case that is difficult to eliminate from the user's perspective - they can't simply make the creation and capture of an accessor conditional.

Reply:
I agree that while unlikely (especially since we pretty much only support `one_to_one` writing access right now), we cannot rule this out in the general case, and so I'd say such a diagnostic is best left for tooling (cc @facuMH).

Reply:
I'm going to continue without the diagnostic then.